src/Compilers/Core/Portable/EncodedStringText.cs

```diff
@@ -86,7 +86,7 @@ private static Encoding GetFallbackEncoding()
             bool canBeEmbedded = false)
         {
             Debug.Assert(stream != null);
-            Debug.Assert(stream.CanRead && stream.CanSeek);
+            Debug.Assert(stream is { CanRead: true, CanSeek: true });
```

Comment:
>Assert [](start = 18, length = 6)

This also subsumes the previous assert which contains a null check.

Reply:
➡️ Filed https://github.com/dotnet/roslyn-analyzers/issues/3029 to discuss.